<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Quick Access to Homologene and Gene Annotation Updates • homologene</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Quick Access to Homologene and Gene Annotation Updates">
<meta property="og:description" content="A wrapper for the homologene database by the National Center for
    Biotechnology Information ('NCBI'). It allows searching for gene homologs across 
    species. Data in this package can be found at &lt;ftp://ftp.ncbi.nih.gov/pub/HomoloGene/build68/&gt;.
    The package also includes an updated version of the homologene database where 
    gene identifiers and symbols are replaced with their latest (at the time of
    submission) version and functions to fetch latest annotation data to keep updated.">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">homologene</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.5.68.19.5.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/oganm/homologene">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="contents col-md-9">

<div id="homologene" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#homologene" class="anchor"></a>homologene</h1></div>

<p>An r package that works as a wrapper to homologene</p>
<p>Available species are</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">homologene<span class="op">::</span>taxData</a></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode R"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co">##    tax_id                      name_txt</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="co">## 1   10090                  Mus musculus</span></a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="co">## 2   10116             Rattus norvegicus</span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="co">## 3   28985          Kluyveromyces lactis</span></a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="co">## 4  318829            Magnaporthe oryzae</span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6"><span class="co">## 5   33169         Eremothecium gossypii</span></a>
<a class="sourceLine" id="cb2-7" data-line-number="7"><span class="co">## 6    3702          Arabidopsis thaliana</span></a>
<a class="sourceLine" id="cb2-8" data-line-number="8"><span class="co">## 7    4530                  Oryza sativa</span></a>
<a class="sourceLine" id="cb2-9" data-line-number="9"><span class="co">## 8    4896     Schizosaccharomyces pombe</span></a>
<a class="sourceLine" id="cb2-10" data-line-number="10"><span class="co">## 9    4932      Saccharomyces cerevisiae</span></a>
<a class="sourceLine" id="cb2-11" data-line-number="11"><span class="co">## 10   5141             Neurospora crassa</span></a>
<a class="sourceLine" id="cb2-12" data-line-number="12"><span class="co">## 11   6239        Caenorhabditis elegans</span></a>
<a class="sourceLine" id="cb2-13" data-line-number="13"><span class="co">## 12   7165             Anopheles gambiae</span></a>
<a class="sourceLine" id="cb2-14" data-line-number="14"><span class="co">## 13   7227       Drosophila melanogaster</span></a>
<a class="sourceLine" id="cb2-15" data-line-number="15"><span class="co">## 14   7955                   Danio rerio</span></a>
<a class="sourceLine" id="cb2-16" data-line-number="16"><span class="co">## 15   8364 Xenopus (Silurana) tropicalis</span></a>
<a class="sourceLine" id="cb2-17" data-line-number="17"><span class="co">## 16   9031                 Gallus gallus</span></a>
<a class="sourceLine" id="cb2-18" data-line-number="18"><span class="co">## 17   9544                Macaca mulatta</span></a>
<a class="sourceLine" id="cb2-19" data-line-number="19"><span class="co">## 18   9598               Pan troglodytes</span></a>
<a class="sourceLine" id="cb2-20" data-line-number="20"><span class="co">## 19   9606                  Homo sapiens</span></a>
<a class="sourceLine" id="cb2-21" data-line-number="21"><span class="co">## 20   9615        Canis lupus familiaris</span></a>
<a class="sourceLine" id="cb2-22" data-line-number="22"><span class="co">## 21   9913                    Bos taurus</span></a></code></pre></div>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/install.packages">install.packages</a></span>(<span class="st">'homologene'</span>)</a></code></pre></div>
<p>or</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">'oganm/homologene'</span>)</a></code></pre></div>
</div>
<div id="usage" class="section level1">
<h1 class="hasAnchor">
<a href="#usage" class="anchor"></a>Usage</h1>
<p>Basic homologene function requires a list of gene symbols or NCBI ids, and an <code>inTax</code> and an <code>outTax</code>. In this example, <code>inTax</code> is the taxon id of <em>mus musculus</em> while <code>outTax</code> is for humans.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw">homologene</span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'Eno2'</span>,<span class="st">'Mog'</span>), <span class="dt">inTax =</span> <span class="dv">10090</span>, <span class="dt">outTax =</span> <span class="dv">9606</span>)</a></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="sourceCode R"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="co">##   10090 9606 10090_ID 9606_ID</span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2"><span class="co">## 1  Eno2 ENO2    13807    2026</span></a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="co">## 2   Mog  MOG    17441    4340</span></a></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw">homologene</span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'Eno2'</span>,<span class="st">'17441'</span>), <span class="dt">inTax =</span> <span class="dv">10090</span>, <span class="dt">outTax =</span> <span class="dv">9606</span>)</a></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="sourceCode R"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="co">##   10090 9606 10090_ID 9606_ID</span></a>
<a class="sourceLine" id="cb8-2" data-line-number="2"><span class="co">## 1  Eno2 ENO2    13807    2026</span></a>
<a class="sourceLine" id="cb8-3" data-line-number="3"><span class="co">## 2   Mog  MOG    17441    4340</span></a></code></pre></div>
<p>For mouse and humans two convenience functions exist that removes the need to provide taxonomic identifiers. Note that the column names are not the same as the <code>homologene</code> output.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw">mouse2human</span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'Eno2'</span>,<span class="st">'Mog'</span>))</a></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="sourceCode R"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1"><span class="co">##   mouseGene humanGene mouseID humanID</span></a>
<a class="sourceLine" id="cb10-2" data-line-number="2"><span class="co">## 1      Eno2      ENO2   13807    2026</span></a>
<a class="sourceLine" id="cb10-3" data-line-number="3"><span class="co">## 2       Mog       MOG   17441    4340</span></a></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="kw">human2mouse</span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'ENO2'</span>,<span class="st">'MOG'</span>,<span class="st">'GZMH'</span>))</a></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="sourceCode R"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="co">##   humanGene mouseGene humanID mouseID</span></a>
<a class="sourceLine" id="cb12-2" data-line-number="2"><span class="co">## 1      ENO2      Eno2    2026   13807</span></a>
<a class="sourceLine" id="cb12-3" data-line-number="3"><span class="co">## 2       MOG       Mog    4340   17441</span></a>
<a class="sourceLine" id="cb12-4" data-line-number="4"><span class="co">## 3      GZMH      Gzmd    2999   14941</span></a>
<a class="sourceLine" id="cb12-5" data-line-number="5"><span class="co">## 4      GZMH      Gzme    2999   14942</span></a>
<a class="sourceLine" id="cb12-6" data-line-number="6"><span class="co">## 5      GZMH      Gzmg    2999   14944</span></a>
<a class="sourceLine" id="cb12-7" data-line-number="7"><span class="co">## 6      GZMH      Gzmf    2999   14943</span></a></code></pre></div>
</div>
<div id="homologenedata2" class="section level1">
<h1 class="hasAnchor">
<a href="#homologenedata2" class="anchor"></a>homologeneData2</h1>
<p>Original homologene database has not been updated since 2014. This package also includes an updated version of the homologene database that replaces gene symbols and identifiers with the their latest version. For the procedure followed for updating, see <a href="https://oganm.com/homologene-update/">this blog post</a> and/or see the <a href="R/updateHomologene.R">processing code</a>.</p>
<p>Using the updated version can help you match genes that cannot matched due to out of date annotations.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw">mouse2human</span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'Mesd'</span>,</a>
<a class="sourceLine" id="cb13-2" data-line-number="2">              <span class="st">'Trp53rka'</span>,</a>
<a class="sourceLine" id="cb13-3" data-line-number="3">              <span class="st">'Cstdc4'</span>,</a>
<a class="sourceLine" id="cb13-4" data-line-number="4">              <span class="st">'Ifit3b'</span>))</a></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="sourceCode R"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="co">## [1] mouseGene humanGene mouseID   humanID  </span></a>
<a class="sourceLine" id="cb14-2" data-line-number="2"><span class="co">## &lt;0 rows&gt; (or 0-length row.names)</span></a></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="kw">mouse2human</span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'Mesd'</span>,</a>
<a class="sourceLine" id="cb15-2" data-line-number="2">              <span class="st">'Trp53rka'</span>,</a>
<a class="sourceLine" id="cb15-3" data-line-number="3">              <span class="st">'Cstdc4'</span>,</a>
<a class="sourceLine" id="cb15-4" data-line-number="4">              <span class="st">'Ifit3b'</span>),</a>
<a class="sourceLine" id="cb15-5" data-line-number="5">            <span class="dt">db =</span> homologeneData2)</a></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode R"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="co">##   mouseGene humanGene mouseID humanID</span></a>
<a class="sourceLine" id="cb16-2" data-line-number="2"><span class="co">## 1      Mesd      MESD   67943   23184</span></a>
<a class="sourceLine" id="cb16-3" data-line-number="3"><span class="co">## 2  Trp53rka    TP53RK  381406  112858</span></a>
<a class="sourceLine" id="cb16-4" data-line-number="4"><span class="co">## 3    Cstdc4      CSTA  433016    1475</span></a>
<a class="sourceLine" id="cb16-5" data-line-number="5"><span class="co">## 4    Ifit3b     IFIT3  667370    3437</span></a></code></pre></div>
<p>The <code>homologeneData2</code> object that comes with the GitHub version of this package is updated weekly but if you are using the CRAN version and want the latest annotations, or if you want to keep a frozen version homologene, you can use the <code>updateHomologene</code> function.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1">homologeneDataVeryNew =<span class="st"> </span><span class="kw">updateHomologene</span>() <span class="co"># update the homologene database with the latest identifiers</span></a>
<a class="sourceLine" id="cb17-2" data-line-number="2"></a>
<a class="sourceLine" id="cb17-3" data-line-number="3"><span class="kw">mouse2human</span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'Mesd'</span>,</a>
<a class="sourceLine" id="cb17-4" data-line-number="4">              <span class="st">'Trp53rka'</span>,</a>
<a class="sourceLine" id="cb17-5" data-line-number="5">              <span class="st">'Cstdc4'</span>,</a>
<a class="sourceLine" id="cb17-6" data-line-number="6">              <span class="st">'Ifit3b'</span>),</a>
<a class="sourceLine" id="cb17-7" data-line-number="7">            <span class="dt">db =</span> homologeneDataVeryNew)</a></code></pre></div>
</div>
<div id="gene-id-syncronization" class="section level1">
<h1 class="hasAnchor">
<a href="#gene-id-syncronization" class="anchor"></a>Gene ID syncronization</h1>
<p>The package also includes functions that were used to create the <code>homologeneData2</code>, for updating outdated gene symbols and identifiers.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(dplyr)</a></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="sourceCode R"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="co">## </span></a>
<a class="sourceLine" id="cb19-2" data-line-number="2"><span class="co">## Attaching package: 'dplyr'</span></a>
<a class="sourceLine" id="cb19-3" data-line-number="3"></a>
<a class="sourceLine" id="cb19-4" data-line-number="4"><span class="co">## The following object is masked from 'package:testthat':</span></a>
<a class="sourceLine" id="cb19-5" data-line-number="5"><span class="co">## </span></a>
<a class="sourceLine" id="cb19-6" data-line-number="6"><span class="co">##     matches</span></a>
<a class="sourceLine" id="cb19-7" data-line-number="7"></a>
<a class="sourceLine" id="cb19-8" data-line-number="8"><span class="co">## The following objects are masked from 'package:stats':</span></a>
<a class="sourceLine" id="cb19-9" data-line-number="9"><span class="co">## </span></a>
<a class="sourceLine" id="cb19-10" data-line-number="10"><span class="co">##     filter, lag</span></a>
<a class="sourceLine" id="cb19-11" data-line-number="11"></a>
<a class="sourceLine" id="cb19-12" data-line-number="12"><span class="co">## The following objects are masked from 'package:base':</span></a>
<a class="sourceLine" id="cb19-13" data-line-number="13"><span class="co">## </span></a>
<a class="sourceLine" id="cb19-14" data-line-number="14"><span class="co">##     intersect, setdiff, setequal, union</span></a></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1">gene_history =<span class="st"> </span><span class="kw">getGeneHistory</span>()</a>
<a class="sourceLine" id="cb20-2" data-line-number="2">oldIds =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">4340964</span>, <span class="dv">4349034</span>, <span class="dv">4332470</span>, <span class="dv">4334151</span>, <span class="dv">4323831</span>)</a>
<a class="sourceLine" id="cb20-3" data-line-number="3">newIds =<span class="st"> </span><span class="kw">updateIDs</span>(oldIds,gene_history)</a>
<a class="sourceLine" id="cb20-4" data-line-number="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/print">print</a></span>(newIds)</a></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="sourceCode R"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1"><span class="co">## [1] "9267698" "4349033" "4332468" "4334150" "4324017"</span></a></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1"><span class="co"># get the latest gene symbols for the ids</span></a>
<a class="sourceLine" id="cb22-2" data-line-number="2"></a>
<a class="sourceLine" id="cb22-3" data-line-number="3">gene_info =<span class="st"> </span><span class="kw">getGeneInfo</span>()</a>
<a class="sourceLine" id="cb22-4" data-line-number="4"></a>
<a class="sourceLine" id="cb22-5" data-line-number="5">gene_info <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb22-6" data-line-number="6"><span class="st">    </span>dplyr<span class="op">::</span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(GeneID <span class="op">%in%</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/integer">as.integer</a></span>(newIds)) <span class="co"># faster to match integers</span></a></code></pre></div>
<div class="sourceCode" id="cb23"><pre class="sourceCode R"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" data-line-number="1"><span class="co">## # A tibble: 5 x 3</span></a>
<a class="sourceLine" id="cb23-2" data-line-number="2"><span class="co">##   tax_id  GeneID Symbol    </span></a>
<a class="sourceLine" id="cb23-3" data-line-number="3"><span class="co">##    &lt;int&gt;   &lt;int&gt; &lt;chr&gt;     </span></a>
<a class="sourceLine" id="cb23-4" data-line-number="4"><span class="co">## 1  39947 4324017 LOC4324017</span></a>
<a class="sourceLine" id="cb23-5" data-line-number="5"><span class="co">## 2  39947 4332468 LOC4332468</span></a>
<a class="sourceLine" id="cb23-6" data-line-number="6"><span class="co">## 3  39947 4334150 LOC4334150</span></a>
<a class="sourceLine" id="cb23-7" data-line-number="7"><span class="co">## 4  39947 4349033 LOC4349033</span></a>
<a class="sourceLine" id="cb23-8" data-line-number="8"><span class="co">## 5  39947 9267698 LOC9267698</span></a></code></pre></div>
</div>
<div id="mishaps" class="section level1">
<h1 class="hasAnchor">
<a href="#mishaps" class="anchor"></a>Mishaps</h1>
<p>As of version version 1.1.68, the output now includes NCBI ids. Since it doesn’t change any of the existing column names or their order, this shouldn’t cause problems in most use cases.</p>
<p>If a you can’t find a gene you are looking for it may have synonyms. See <a href="https://github.com/oganm/geneSynonym.git">geneSynonym</a> package to find them. If you have other problems open an issue or send a mail.</p>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Download from CRAN at <br><a href="https://cloud.r-project.org/package=homologene">https://​cloud.r-project.org/​package=homologene</a>
</li>
<li>Browse source code at <br><a href="https://github.com/oganm/homologene">https://​github.com/​oganm/​homologene</a>
</li>
<li>Report a bug at <br><a href="https://github.com/oganm/homologene/issues">https://​github.com/​oganm/​homologene/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Ogan Mancarci <br><small class="roles"> Author, maintainer </small>  </li>
<li><a href="authors.html">All authors...</a></li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://travis-ci.org/oganm/homologene"><img src="https://travis-ci.org/oganm/homologene.svg?branch=master" alt="Build Status"></a></li>
<li><a href="https://codecov.io/gh/oganm/homologene"><img src="https://codecov.io/gh/oganm/homologene/branch/master/graph/badge.svg" alt="codecov"></a></li>
<li><a href="https://cran.r-project.org/package=homologene"><img src="https://www.r-pkg.org/badges/version/homologene?color=#32BD36"></a></li>
<li><a href="https://github.com/oganm/homologene"><img src="https://img.shields.io/badge/devel%20version-1.4.68.19.4.21-blue.svg"></a></li>
</ul>
</div>
</div>
</div>

      <footer><div class="copyright">
  <p>Developed by Ogan Mancarci.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
